### PR TITLE
Add users api

### PR DESCRIPTION
**Pull request by Astel**

### CODE REVIEW

File: src/main/java/com/epam/lab/payments/web/AuthorizationController.java [hunk 1]
```diff
@@ -44,16 +46,34 @@ public ModelAndView createNewUser(@Valid UserEntity user, BindingResult bindingR
...
         return modelAndView;
     }
 
     @RequestMapping(value={"/", LOGIN}, method = RequestMethod.GET)
-    public ModelAndView login(){
+    public ModelAndView login(HttpServletRequest request){
         ModelAndView modelAndView = new ModelAndView();
-        modelAndView.setViewName(LOGIN);
+
+        Principal principal = request.getUserPrincipal();
```
Comment (iliaonishchenko) [inline src/main/java/com/epam/lab/payments/web/AuthorizationController.java]: Use Optional here.

File: src/main/java/com/epam/lab/payments/services/AuthorizationService.java [hunk 2]
```diff
@@ -10,14 +12,29 @@
...
-        userEntity.setPassword(bCryptPasswordEncoder.encode(userEntity.getPassword()));
-        userRepository.save(userEntity);
+    public void save(UserDTO userDTO) {
+        userDTO.setPassword(bCryptPasswordEncoder.encode(userDTO.getPassword()));
+        userRepository.save(userMapper.userDtoToUser(userDTO));
     }
 
-    public UserEntity findUserByEmail(String email) {
-        return userRepository.findByEmail(email);
+    public UserDTO findUserByEmail(String email) {
```
Comment (iliaonishchenko) [inline src/main/java/com/epam/lab/payments/services/AuthorizationService.java]: It can be null here, so use Optional.

**Review by iliaonishchenko — COMMENTED** [review]
Review done.

File: src/main/java/com/epam/lab/payments/services/AuthorizationService.java (hunk 2 above)
Comment (Astel) [reply src/main/java/com/epam/lab/payments/services/AuthorizationService.java]: done

File: src/main/java/com/epam/lab/payments/web/AuthorizationController.java (hunk 1 above)
Comment (Astel) [reply src/main/java/com/epam/lab/payments/web/AuthorizationController.java]: rework to isAuthenticated()

File: src/main/java/com/epam/lab/payments/UserValidator.java [hunk 3]
```diff
@@ -18,23 +18,23 @@
...
-        UserEntity userEntity = (UserEntity) o;
+        UserDTO userDTO = (UserDTO) o;
 
         ValidationUtils.rejectIfEmptyOrWhitespace(errors, EMAIL, "NotEmpty");
-        if (userEntity.getEmail().length() < 6 || userEntity.getEmail().length() > 32) {
+        if (userDTO.getEmail().length() < 6 || userDTO.getEmail().length() > 32) {
             errors.reject(EMAIL, "Size.userForm.email");
         }
-        if (authorizationService.findUserByEmail(userEntity.getEmail()) != null) {
+        if (authorizationService.findUserByEmail(userDTO.getEmail()) != null) {
```
Comment (iliaonishchenko) [inline src/main/java/com/epam/lab/payments/UserValidator.java]: Condition always true, because Optional is returned by findUserByEmail.

**Review by iliaonishchenko — APPROVED** [review]
Please fix one issue and you are free to merge.

**Pull request merged**